src/tickit/cli.py

```diff
@@ -33,19 +33,24 @@ def main(ctx: Context, log_level: str):
 
 
 @main.command(help="run one or more components without a scheduler")
-@click.argument("components", nargs=-1)
+@click.argument("components", required=True, nargs=-1)
 @click.argument("config_path")
 @click.option("--backend", default="kafka", type=click.Choice(list(interfaces(True))))
-def components(config_path: str, components: Set[ComponentID], backend: str) -> None:
+def components(
+    config_path: str,
+    components: Optional[Union[Set[ComponentID], Literal["all"]]],
```

Comment:
I don't know how well click handles this. How does it determine between a set containing 'all' and the literal 'all'? It looks like it's completely ignoring the type anyway and passing a tuple (which answers the comment below).

Reply:
Yeah click doesn't really handle it at all, it seams to just give me a tuple of stuff. I think a answer would be using a click.option, but it doesn't really fit the command to then have the list in a option flag.